models/eprop_iaf_bsshslm_2020.cpp

```diff
@@ -0,0 +1,488 @@
+/*
+ *  eprop_iaf_bsshslm_2020.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// nest models
+#include "eprop_iaf_bsshslm_2020.h"
+
+// C++
+#include <limits>
+
+// libnestutil
+#include "dict_util.h"
+#include "numerics.h"
+
+// nestkernel
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "nest_impl.h"
+#include "universal_data_logger_impl.h"
+
+// sli
+#include "dictutils.h"
+
+namespace nest
+{
+
+void
+register_eprop_iaf_bsshslm_2020( const std::string& name )
+{
+  register_node_model< eprop_iaf_bsshslm_2020 >( name );
+}
+
+/* ----------------------------------------------------------------
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+RecordablesMap< eprop_iaf_bsshslm_2020 > eprop_iaf_bsshslm_2020::recordablesMap_;
+
+template <>
+void
+RecordablesMap< eprop_iaf_bsshslm_2020 >::create()
+{
+  insert_( names::learning_signal, &eprop_iaf_bsshslm_2020::get_learning_signal_ );
+  insert_( names::surrogate_gradient, &eprop_iaf_bsshslm_2020::get_surrogate_gradient_ );
+  insert_( names::V_m, &eprop_iaf_bsshslm_2020::get_v_m_ );
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors for parameters, state, and buffers
+ * ---------------------------------------------------------------- */
+
+eprop_iaf_bsshslm_2020::Parameters_::Parameters_()
+  : C_m_( 250.0 )
+  , c_reg_( 0.0 )
+  , E_L_( -70.0 )
+  , f_target_( 0.01 )
+  , gamma_( 0.3 )
+  , I_e_( 0.0 )
+  , psc_scale_factor_( "alpha_complement" )
```

Comment:
See comment on adapt variant.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/2867/commits/c612b0771fc2f2c6d8f0248e09eda043cd5f6119